orangecontrib/timeseries/widgets/owtabletotimeseries.py

```diff
@@ -26,100 +57,255 @@ class Inputs:
+            self.commit.now()
+            return
+
+        with signal_blocking(self.var_combo):
+            model = self.var_combo.model()
+            self.data = data
+            time_vars = [
+                var for var in chain(data.domain.variables, data.domain.metas)
+                if var.is_time]
+            cont_vars = [
+                var for var in chain(data.domain.variables, data.domain.metas)
+                if var.is_continuous and not var.is_time]
+            separator = [VariableListModel.Separator
+                         ] if len(time_vars) > 1 and cont_vars else []
+            model[:] = time_vars + separator + cont_vars
+
+            self.controls.implied_sequence.buttons[0].setDisabled(not model)
+            if not model:
+                self.implied_sequence = 1
+            if model and self.attribute not in [var.name
+                                                for var in time_vars + cont_vars]:
+                self.attribute = getattr(data, "time_variable", model[0]).name
         self.commit.now()
 
     @gui.deferred
     def commit(self):
+        self.Warning.clear()
+        self.Error.year_out_of_range.clear()
+        self.Error.invalid_time.clear()
+        if not self.data:
+            ts = None
+        elif self.implied_sequence == 0:
+            ts = self._series_from_variable()
+        else:
+            ts = self._series_by_sequence()
+        self.Outputs.time_series.send(ts)
+
+    def _series_from_variable(self):
         data = self.data
-        self.Information.clear()
-        if not data:
-            self.Outputs.time_series.send(None)
-            return
+        time_var = data.domain[self.attribute]
+        ts = Timeseries.make_timeseries_from_continuous_var(data, time_var)
+        self.Warning.nan_times(self.attribute,
+                               shown=ts is None or len(data) != len(ts))
+        return ts
 
-        if self.order is None:
-            ts = Timeseries.make_timeseries_from_sequence(data)
-        else:
-            ts = Timeseries.make_timeseries_from_continuous_var(data, self.order)
-            # Warn if instances are omitted because of nans in selected attr
-            times, sparse = data.get_column_view(self.order)
-            if sparse:
-                times = times.data
+    def _series_by_sequence(self):
+        if not self.is_valid_time():
+            self.Error.invalid_time()
+            return None
+
+        start = datetime.datetime(*self.get_time(), 0, datetime.timezone.utc)
+        delta = int(self.steps) * StepOptions[self.unit].delta
+        try:
+            ts = Timeseries.make_timeseries_from_sequence(
+                self.data, delta, start,
+                have_date=self.use_date, have_time=self.use_time)
+        except ValueError as exc:
+            msg = str(exc)
+            if msg.startswith("year") and msg.endswith("out of range"):
+                self.Error.year_out_of_range(msg.capitalize())
```

Comment:
Writing 11112022 into the year Line Edit is not handled.

Reply:
Fixed and tested.